app/code/core/Mage/Payment/Helper/Data.php

```diff
@@ -120,9 +120,20 @@ public function getMethodFormBlock(Mage_Payment_Model_Method_Abstract $method)
      *
      * @return  Mage_Core_Block_Template|Mage_Core_Block_Abstract
      */
-    public function getInfoBlock(Mage_Payment_Model_Info $info)
+    public function getInfoBlock(Mage_Payment_Model_Info|false $info)
     {
-        $blockType = $info->getMethodInstance()->getInfoBlockType();
+        $method = $info !== false ? $info->getMethod() : $this->__('No Data Found');
```

Comment:
```suggestion
        $method = $info !== false ? $info->getMethod() : $this->__('No Data Found');
```

same here, I usually prefer not to use "negative ifs" (like $info !== false) because they're harder to read

also, "No Data Found" would be a new string to add to translation and I'd avoid that, using "N/A" (which is already in translation) or just a "-" or something like that

Reply:
also, where is this showing? I can't find it in frontend of backend

Reply:
The condition is swapped now, so that the `!==` is no longer necessary.

I have specifically chosen a string here that might fit best and already has a translation (see `app/locale/en_US/Mage_Adminhtml.csv:659` for `No Data Found`). When I should use different string, please let me know (and which one exactly). It's not that important to me which one is used here.

To see the string, just place an order, go to `sales_flat_order_payment` table in your database and delete the row for your order. Then you will see it in the frontend as well as in the adminhtml.

Reply:
my fault, sorry! I somehow searched for "no data found" and didn't find it in the translations csv :-\ weird